tio/bufferedreader.go

```diff
@@ -257,6 +272,55 @@ func (buffer *BufferedReader) parseMLE64() ([]byte, int) {
 	return buffer.extractMessage(int(messageLen), buffer.paramMLE+8)
 }
 
+// messages are separated by regexp
+func (buffer *BufferedReader) parseDelimiterMultiLine() ([]byte, int) {
+	startRegexp := regexp.MustCompile(buffer.delimiterStr)
```

Comment:
move this to the init function and store the compiled regex for improved performance (see above)